image/image.go

```diff
@@ -70,28 +75,49 @@ func (i *genericImage) Signatures() ([][]byte, error) {
 }
 
 func (i *genericImage) Inspect() (*types.ImageInspectInfo, error) {
-	// TODO(runcom): unused version param for now, default to docker v2-1
-	m, err := i.getSchema1Manifest()
+	m, err := i.getParsedManifest()
 	if err != nil {
 		return nil, err
 	}
-	ms1, ok := m.(*manifestSchema1)
-	if !ok {
-		return nil, fmt.Errorf("error retrivieng manifest schema1")
-	}
-	v1 := &v1Image{}
-	if err := json.Unmarshal([]byte(ms1.History[0].V1Compatibility), v1); err != nil {
-		return nil, err
+	switch t := m.(type) {
+	case *manifestSchema1:
+		v1 := &v1Image{}
+		if err := json.Unmarshal([]byte(t.History[0].V1Compatibility), v1); err != nil {
+			return nil, err
+		}
+		return &types.ImageInspectInfo{
+			Tag:           t.Tag,
+			DockerVersion: v1.DockerVersion,
+			Created:       v1.Created,
+			Labels:        v1.Config.Labels,
+			Architecture:  v1.Architecture,
+			Os:            v1.OS,
+			Layers:        t.GetLayers(),
+		}, nil
+	case *manifestSchema2:
+		rawConfig, _, err := i.src.GetBlob(t.Config.Digest)
+		if err != nil {
+			return nil, err
+		}
+		defer rawConfig.Close()
+		config, err := ioutil.ReadAll(rawConfig)
+		if err != nil {
+			return nil, err
+		}
+		v1 := &v1Image{}
+		if err := json.Unmarshal(config, v1); err != nil {
+			return nil, err
+		}
+		return &types.ImageInspectInfo{
+			DockerVersion: v1.DockerVersion,
+			Created:       v1.Created,
+			Labels:        v1.Config.Labels,
+			Architecture:  v1.Architecture,
+			Os:            v1.OS,
+			Layers:        t.GetLayers(),
+		}, nil
 	}
-	return &types.ImageInspectInfo{
-		Tag:           ms1.Tag,
-		DockerVersion: v1.DockerVersion,
-		Created:       v1.Created,
-		Labels:        v1.Config.Labels,
-		Architecture:  v1.Architecture,
-		Os:            v1.OS,
-		Layers:        ms1.GetLayers(),
-	}, nil
+	return nil, errors.New("error inspecting image")
```

Comment:
This should say a bit more about what kind of error or what the user is supposed to do. (Unsupported manifest type? Internal error unexpected type inconsistency?) Or, with the move to a `genericManifest` method, this line could perhaps entirely disappear.